gitopsaddon/gitopsaddon_install.go

```
@@ -393,8 +425,117 @@ func (r *GitopsAddonReconciler) recoverFromFailedWebhookInstallPlan(ctx context.
+//  sees installedCSV set in the subscription status and considers the operator installed,
+//  so it does NOT issue a new InstallPlan. The addon is stuck waiting for the operator
+//  deployment which will never appear.
+func (r *GitopsAddonReconciler) recoverFromGhostCompleteInstallPlan(ctx context.Context) error {
+	subName := getOLMEnvOrDefault("OLM_SUBSCRIPTION_NAME", "openshift-gitops-operator")
+	subNamespace := getOLMEnvOrDefault("OLM_SUBSCRIPTION_NAMESPACE", GitOpsOperatorNamespace)
+
+	existing := &unstructured.Unstructured{}
+	existing.SetAPIVersion("operators.coreos.com/v1alpha1")
+	existing.SetKind("Subscription")
+	if err := r.Get(ctx, types.NamespacedName{Name: subName, Namespace: subNamespace}, existing); err != nil {
+		return nil // subscription not found yet — nothing to recover
```

golangci-lint (2.12.2)

[error] error is not nil (line 459) but it returns nil

(nilerr)